blocks/variables.js

```diff
@@ -186,22 +186,21 @@ Blockly.Blocks['variables_get'] = {
       }
 
       if(topBlock.index != null){
-        // find the type of the child that has a type and set output to that type
+        // find the type of the connection that has a type and set output to that type
         var checkType;
-        var children = variableUses[topBlock.index].getChildren();
-        if(children.length == 0){
-          this.modifyBlock("None");
+
+        try{checkType = variableUses[topBlock.index].inputList[0]
+                                                    .connection
+                                                    .targetConnection
+                                                    .check_[0];
+        }catch(e){}
```

Comment:
It might greatly benefit dev & testing if we'd `console.error(e)` or etc. in all these `catch` blocks. Unless you're eating errors to patch over not having types for all blocks or something like that.  

Either way, my comment shouldn't block merging this PR but maybe something to consider hitting at some point.